web/packages/teleport/src/Discover/Kubernetes/HelmChart/HelmChart.tsx

```diff
@@ -364,17 +370,23 @@ const generateCmd = (data: {
 
   const yamlRoles = roles.join(',').toLowerCase();
 
+  const joinLabelsText = data.joinLabels
+    ? data.joinLabels.map(l => `    ${l.name}: ${l.value}`).join('\n')
+    : '';
+
   return `cat << EOF > prod-cluster-values.yaml
 roles: ${yamlRoles}
 authToken: ${data.tokenId}
 proxyAddr: ${data.proxyAddr}
 kubeClusterName: ${data.clusterName}
 labels:
     teleport.internal/resource-id: ${data.resourceId}
+${joinLabelsText}
```

Comment:
hrm, this will be a newline if this is empty, can we add it conditionally on line 383 instead? also i'd probably write a test for this func to make sure our yaml output is expected on different inputs

(btw did you test that the non-aws kube flow also works after all these changes?)

Reply:
I've removed empty newline 👍 https://github.com/gravitational/teleport/pull/36634/commits/65c9c4778e6b52540192ed11938cfc586a6baf5c

Will write a test a bit later.